IOTConnectSDK/iotconnect-sdk_2_1/src/main/java/com/iotconnectsdk/utils/SecurityHelper.kt

```diff
@@ -0,0 +1,184 @@
+package com.iotconnectsdk.utils
+
+
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import org.bouncycastle.util.io.pem.PemReader
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileReader
+import java.io.InputStream
+import java.security.KeyFactory
+import java.security.KeyStore
+import java.security.NoSuchAlgorithmException
+import java.security.PrivateKey
+import java.security.SecureRandom
+import java.security.Security
+import java.security.cert.Certificate
+import java.security.cert.CertificateFactory
+import java.security.cert.X509Certificate
+import java.security.spec.PKCS8EncodedKeySpec
+import javax.net.ssl.KeyManagerFactory
+import javax.net.ssl.SSLContext
+import javax.net.ssl.SSLSocketFactory
+import javax.net.ssl.TrustManagerFactory
+
+/**
+ * @Author uttam.kachhad
+ * @Create 16/03/2022
+ * @Description
+ */
+internal object SecurityHelper {
+    fun createSocketFactory(
+        caFile: String?,
+        clientCrtFile: String?,
+        clientKeyFile: String?,
+        clientKeyPassword: String,
+        companySID: String?,
+        deviceUniqueID: String?
+    ): SSLSocketFactory? {
+
+        // Creates a TLS socket factory with the given
+        // CA certificate file, client certificate, client key
+        // In this case, we are working without a client key password
+        return try {
+            Security.addProvider(BouncyCastleProvider())
+            val keyManagers = createKeyManagerFactory(
+                clientCrtFile,
+                clientKeyFile,
+                clientKeyPassword,
+                companySID,
+                deviceUniqueID
+            )!!.keyManagers
```

Comment:
@sw-tt-tusharojha @sw-tt-rajkiranvank !! will create a 'NullPointerException'  can you please write in better way

Reply:
@sw-ot-deveshmevada 
Issue resolved **!!**  change to this **?.**, now NullPointerException will not be created